NitroxServer/ConsoleCommands/HelpCommand.cs

```diff
@@ -0,0 +1,65 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using NitroxModel.DataStructures.GameLogic;
+using NitroxModel.DataStructures.Util;
+using NitroxModel.Logger;
+using NitroxServer.ConsoleCommands.Abstract;
+using NitroxServer.ConsoleCommands.Processor;
+using NitroxServer.GameLogic.Players;
+
+namespace NitroxServer.ConsoleCommands
+{
+    internal class HelpCommand : Command
+    {
+        private readonly PlayerData playerData;
+        private Dictionary<Perms, List<string>> helpTexts = new Dictionary<Perms, List<string>>();
+
+        public HelpCommand(PlayerData playerData) : base("help", Perms.PLAYER, "", "Display help about supported commands")
+        {
+            this.playerData = playerData;
+        }
+
+        public override void RunCommand(string[] args, Optional<Player> player)
+        {
+            if (player.IsPresent())
+            {
+                List<string> cmdsText = GetHelpText(playerData.GetPermissions(player.Get().Name));
+                cmdsText.ForEach(cmdText => SendServerMessageIfPlayerIsPresent(player, cmdText));
+            }
+            else
+            {
+                List<string> cmdsText = GetHelpText(Perms.CONSOLE);
+                cmdsText.ForEach(cmdText => Log.Info(cmdText));
+            }
+        }
+
+        public override bool VerifyArgs(string[] args)
+        {
+            return args.Length == 0;
+        }
+        
+        private class CommandComparer : IComparer<Command>
+        {
+            public int Compare(Command x, Command y)
+            {
+                return x.Name.CompareTo(y.Name);
+            }
+        }
+
+        private List<string> GetHelpText(Perms perm)
+        {
+            List<string> helpText;
+            if (helpTexts.TryGetValue(perm, out helpText))
```

Comment:
Do we need to cache?  Seems like something infrequently run so less code / complexity is preferred over optimization.

Reply:
i glanced through other command class found they pretty lightweightted. except that im unsure about the cost of NitrozServiceLocator. The cache is introduced to reduce attack surface at the initial purpose. Its okay to remove cache if you are sure its better.
Second confirmation please.

Reply:
never mind. i removed it.